### PR TITLE
modules: Add QRCode module.

### PR DESCRIPTION
**Pull request by betterengineering**
# Overview
We want to create apps that include QR codes and wanted to offer a better experience then embedding an image. This change adds a new module, `qrcode` that allows for generating QR codes that look great on the display.

# Notes on QR Codes
- QR Codes have version numbers, which roughly equates to the size of the code in terms of _modules_
- A module is a 1x1 square, so for us a module is a pixel
- Versions are from 1-40. We can use the following versions on a Tidbyt:
    - 1 == 21x21
    - 2 == 25x25
    - 3 == 29x29
- One gotcha is that lower version numbers also can contain less data. This becomes a problem if a users wants to include large URLs in small QRCodes without shortening them
- QRCodes have a an error correction feature, which is useful in environments like a paper menu when there might be smudges, etc.
    - Because we're using small QRCodes, we can't support a super high level of error correction without sacrificing the length of URL we have

# Changes
- [modules: Add QRCode module.](https://github.com/tidbyt/pixlet/commit/e01b5534f2259c67f4ac116a282d56c67c9f8e97)
    - This commit adds a module to generate QR codes to pixlet! This native QR code integration can be used to create promotional apps.
    
# Example
```starlark
load("render.star", "render")
load("qrcode.star", "qrcode")

def main(config):
    code = qrcode.generate(
        url = "https://tidbyt.com?utm_source=pixlet_example",
        size = "large",
        color = "#fff",
        background = "#000",
    )

    return render.Root(
        child = render.Padding(
            child = render.Image(src = code),
            pad = 1,
        ),
    )
```

### CODE REVIEW

File: runtime/modules/qrcode/qrcode.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+)
+
+var (
+	once   sync.Once
+	module starlark.StringDict
+)
+
+func LoadModule() (starlark.StringDict, error) {
+	once.Do(func() {
+		rand.Seed(time.Now().UnixNano())
```
Comment (matslina) [inline runtime/modules/qrcode/qrcode.go:28]: Why `rand.Seed()`?

File: runtime/modules/qrcode/qrcode.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		"size", &starSize,
+		"color?", &starColor,
+		"background?", &starBackground,
+	); err != nil {
+		return nil, fmt.Errorf("unpacking arguments for generate: %w", err)
+	}
+
+	// Validate size.
+	if !contains([]string{"small", "medium", "large"}, starSize.GoString()) {
+		return nil, fmt.Errorf("size must be small, medium, or large")
```
Comment (matslina) [inline runtime/modules/qrcode/qrcode.go:63]: Could be the `default:` of the switch below.

File: runtime/modules/qrcode/qrcode.go (hunk 1 above)
Comment (betterengineering) [reply runtime/modules/qrcode/qrcode.go]: Oops! Bad copy/paste. Fixed

File: examples/qrcode.star [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+def main(config):
+    url = "https://tidbyt.com?utm_source=pixlet_example"
+    code = cache.get(url)
+    if code == None:
+        code = qrcode.generate(
+            url = url,
+            size = "large",
+            color = "#fff",
+            background = "#000",
+        )
```
[the remark this answers is not in the document]
Comment (betterengineering) [reply examples/qrcode.star:16]: I updated the code with a base64 encoding of the image, thanks for catching!

File: runtime/modules/qrcode/qrcode.go (hunk 2 above)
Comment (betterengineering) [reply runtime/modules/qrcode/qrcode.go:60]: I'd hate for a bad value to default though, which could end up being confusing to the user

Comment (betterengineering) [reply runtime/modules/qrcode/qrcode.go:60]: Ah! You mean error on default. I like it. One sec

Comment (matslina) [reply runtime/modules/qrcode/qrcode.go:60]: I meant that the `contains()` method is effectively already implemented by the switch statement. This check could be a default clause in the switch.

**Review by matslina — APPROVED** [review]

**Pull request merged**